crates/lair_keystore_api/src/sodium_secretstream.rs

```diff
@@ -192,72 +224,86 @@ where
 pub fn new_s3_client<T, S, R>(
     send: S,
     recv: R,
-    srv_id_pub_key: sodoken::BufReadSized<32>,
+    srv_id_pub_key: BinDataSized<32>,
 ) -> impl Future<Output = LairResult<(S3Sender<T>, S3Receiver<T>)>> + 'static + Send
 where
     T: 'static + serde::Serialize + for<'de> serde::Deserialize<'de> + Send,
     S: 'static + tokio::io::AsyncWrite + Send + Unpin,
     R: 'static + tokio::io::AsyncRead + Send + Unpin,
 {
     async move {
-        use sodoken::crypto_box::curve25519xchacha20poly1305 as cbox;
-        use sodoken::kx;
-
         // box these up into trait objects so we can easily refer to their types.
         let mut send: PrivRawSend = Box::new(send);
         let mut recv: PrivRawRecv = Box::new(recv);
 
         // generate an ephemeral cbox keypair
-        let eph_cbox_pub = sodoken::BufWriteSized::new_no_lock();
-        let eph_cbox_sec = sodoken::BufWriteSized::new_mem_locked()?;
-        cbox::keypair(eph_cbox_pub.clone(), eph_cbox_sec.clone()).await?;
+        let mut eph_cbox_pub = [0; sodoken::crypto_box::XSALSA_PUBLICKEYBYTES];
+        let mut eph_cbox_sec = sodoken::SizedLockedArray::<
+            { sodoken::crypto_box::XSALSA_SECRETKEYBYTES },
+        >::new()?;
+        // TODO Should be a crypto_kx_keypair?
+        sodoken::crypto_box::xsalsa_keypair(
+            &mut eph_cbox_pub,
+            &mut eph_cbox_sec.lock(),
+        )?;
 
         // generate an ephemeral kx keypair
-        let eph_kx_pub = sodoken::BufWriteSized::new_no_lock();
-        let eph_kx_sec = sodoken::BufWriteSized::new_mem_locked()?;
-        kx::keypair(eph_kx_pub.clone(), eph_kx_sec.clone())?;
+        let mut eph_kx_pub = [0; sodoken::crypto_box::XSALSA_PUBLICKEYBYTES];
+        let mut eph_kx_sec = sodoken::SizedLockedArray::<
+            { sodoken::crypto_box::XSALSA_SECRETKEYBYTES },
+        >::new()?;
+        sodoken::crypto_box::xsalsa_keypair(
```

Comment:
Same here, we want `kx` keypairs

Reply:
The function that this was calling isn't exposed through sodoken any more, should it be?